pkg/vault/test/foundry/BatchRouterERC4626Pool.t.sol

```diff
@@ -0,0 +1,504 @@
+        router.addLiquidityToBuffer(waUSDC, bufferInitialAmount, bufferInitialAmount, lp);
+        vm.stopPrank();
+    }
+
+    function initializeERC4626Pool() private {
+        TokenConfig[] memory tokenConfig = new TokenConfig[](2);
+        tokenConfig[waDaiIdx].token = IERC20(waDAI);
+        tokenConfig[waUsdcIdx].token = IERC20(waUSDC);
+        tokenConfig[0].tokenType = TokenType.WITH_RATE;
+        tokenConfig[1].tokenType = TokenType.WITH_RATE;
+        tokenConfig[waDaiIdx].rateProvider = IRateProvider(address(waDAI));
+        tokenConfig[waUsdcIdx].rateProvider = IRateProvider(address(waUSDC));
+
+        PoolMock newPool = new PoolMock(IVault(address(vault)), "ERC4626 Pool", "ERC4626P");
+
+        factoryMock.registerTestPool(address(newPool), tokenConfig, poolHooksContract);
+
+        vm.label(address(newPool), "erc4626 pool");
+        erc4626Pool = address(newPool);
+
+        vm.startPrank(bob);
+        waDAI.approve(address(permit2), MAX_UINT256);
+        permit2.approve(address(waDAI), address(router), type(uint160).max, type(uint48).max);
+        waUSDC.approve(address(permit2), MAX_UINT256);
+        permit2.approve(address(waUSDC), address(router), type(uint160).max, type(uint48).max);
+
+        dai.mint(bob, erc4626PoolInitialAmount);
+        dai.approve(address(waDAI), erc4626PoolInitialAmount);
+        waDAI.deposit(erc4626PoolInitialAmount, bob);
+
+        usdc.mint(bob, erc4626PoolInitialAmount);
+        usdc.approve(address(waUSDC), erc4626PoolInitialAmount);
+        waUSDC.deposit(erc4626PoolInitialAmount, bob);
+
+        _initPool(
+            erc4626Pool,
+            [erc4626PoolInitialAmount, erc4626PoolInitialAmount].toMemoryArray(),
+            erc4626PoolInitialBPTAmount - MIN_BPT
+        );
+
+        IERC20(address(erc4626Pool)).approve(address(permit2), MAX_UINT256);
+        permit2.approve(address(erc4626Pool), address(router), type(uint160).max, type(uint48).max);
+        permit2.approve(address(erc4626Pool), address(batchRouter), type(uint160).max, type(uint48).max);
+
+        IERC20(address(erc4626Pool)).approve(address(router), type(uint256).max);
+        IERC20(address(erc4626Pool)).approve(address(batchRouter), type(uint256).max);
+
+        vm.stopPrank();
+    }
+
+    modifier checkBuffersWhenStaticCall(address sender) {
+        uint256 beforeUSDCBalance = usdc.balanceOf(sender);
+        uint256 beforeDAIBalance = dai.balanceOf(sender);
+        (uint256 beforeWaUSDCBufferBalanceUnderlying, uint256 beforeWaUSDCBufferBalanceWrapped) = vault
+            .getBufferBalance(waUSDC);
+        (uint256 beforeWaDAIBufferBalanceUnderlying, uint256 beforeWaDAIBufferBalanceWrapped) = vault.getBufferBalance(
+            waDAI
+        );
+
+        _;
+
+        uint256 afterUSDCBalance = usdc.balanceOf(sender);
+        assertEq(beforeUSDCBalance - afterUSDCBalance, 0, "USDC balance should be the same");
```

Comment:
nit (here and elsewhere): if it's just before == after (i.e. no extra operation between before and after), it reads a bit easier like this:
```suggestion
        assertEq(beforeUSDCBalance, afterUSDCBalance, "USDC balance should be the same");
```